prime-router/metadata/fhir_mapping/hl7/datatypes/CX/Identifier.yml

```diff
@@ -0,0 +1,231 @@
+resourceType: Identifier
+
+value:
+  type: STRING
+  valueOf: CX.1
+  expressionType: HL7Spec
+
+extension_1:
+  generateList: true
+  expressionType: nested
+  expressionsMap:
+    url:
+      type: SYSTEM_URL
+      value: identifier-check-digit
+    valueString:
+      type: STRING
+      expressionType: HL7Spec
+      valueOf: CX.2
+
+extension_2:
+  generateList: true
+  expressionType: nested
+  expressionsMap:
+    url:
+      type: SYSTEM_URL
+      value: naming-system-check-digit
+    valueString:
+      type: STRING
+      expressionType: HL7Spec
+      valueOf: CX.3
+
+# Checking if HD is an Organization or a System by looking at the ID type.
+# If ID type is null or ID type is not ISO or UUID, treat as organization.
+# Otherwise, treat as system.
+assigner_1:
+  vars:
+    cx43: String, CX.4.3
+  condition: $cx43 NULL
+  valueOf: datatypes/HD/Organization
+  expressionType: reference
+  specs: CX.4
+
+assigner_2:
+  vars:
+    cx43: String, CX.4.3
+  condition: $cx43 NOT_NULL && $cx43 NOT_EQUALS ISO && $cx43 NOT_EQUALS UUID
+  valueOf: datatypes/HD/Organization
+  expressionType: reference
+  specs: CX.4
+
+system:
+  vars:
+    cx41: String, CX.4.1
+    cx42: String, CX.4.2
+    cx43: String, CX.4.3
+  condition: $cx43 EQUALS ISO || $cx43 EQUALS UUID
+  expressionType: nested
+  expressions:
+    - condition: $cx41 NOT_NULL
+      type: STRING
+      valueOf: $cx41
+    - condition: $cx41 NULL && $cx43 EQUALS ISO
+      constants:
+        prefix: "urn:oid:"
+      vars:
+        systemURI: $prefix + CX.4.2
+      type: STRING
+      valueOf: $systemURI
+    - condition: $cx41 NULL && $cx43 EQUALS UUID
+      constants:
+        prefix: "urn:uuid:"
+      vars:
+        systemURI: $prefix + CX.4.2
+      type: STRING
+      valueOf: $systemURI
+_system:
+  vars:
+    cx41: String, CX.4.1
+    cx42: String, CX.4.2
+    cx43: String, CX.4.3
+  condition: $cx43 EQUALS ISO || $cx43 EQUALS UUID
+  expressionType: nested
+  expressionsMap:
+    extension_1:
```

Comment:
Can we either give these real names or collapse them into an `expressions` and just make it a list?

Reply:
This actually required the names to signify `extension`, otherwise nothing appeared in the output. So I made the extension names more specific, but left this as an `expressionsMap`